pkg/controller/service/controller.go

```diff
@@ -649,6 +663,38 @@ func getNodeConditionPredicate() NodeConditionPredicate {
 	}
 }
 
+func shouldSyncNode(oldNode, newNode *v1.Node) bool {
+	if oldNode.Spec.Unschedulable != newNode.Spec.Unschedulable {
+		return true
+	}
+
+	if !reflect.DeepEqual(oldNode.Labels, newNode.Labels) {
+		return true
+	}
+
+	oldNodeReadyStatus := nodeReadyConditionStatus(oldNode)
+	newNodeReadyStatus := nodeReadyConditionStatus(newNode)
+
+	// ready conditions unknown for both nodes, skip reconcile
+	if oldNodeReadyStatus == "" && newNodeReadyStatus == "" {
+		return false
```

Comment:
nit: isn't this the same as `return oldNodeReadyStatus != newNodeReadyStatus`?

Reply:
+1

Reply:
fixed